src/controllers/connectionManager.ts

```diff
@@ -600,6 +601,35 @@ export default class ConnectionManager {
             // close active connection
             await this.disconnect(fileUri);
             // connect to the server/database
+            if (connectionCreds.authenticationType === Constants.azureMfa) {
+                if (!connectionCreds.azureAccountToken) {
+                    let account = this.accountStore.getAccount(connectionCreds.accountId);
+                    let profile = new ConnectionProfile();
+                    profile.accountId = connectionCreds.accountId;
+                    profile.authenticationType = connectionCreds.authenticationType;
+                    profile.azureAccountToken = connectionCreds.azureAccountToken;
+                    profile.database = connectionCreds.database;
+                    profile.email = connectionCreds.email;
+                    profile.password = connectionCreds.password;
+                    profile.server = connectionCreds.server;
+                    let azureAccountToken = await this.azureController.refreshToken(account, this.accountStore, providerSettings.resources.databaseResource);
+                    if (!azureAccountToken) {
+                        let errorMessage = LocalizedConstants.msgAccountRefreshFailed;
+                        await this.vscodeWrapper.showErrorMessage(
+                            errorMessage, LocalizedConstants.refreshTokenLabel).then(async refreshResult => {
+                            if (refreshResult === LocalizedConstants.refreshTokenLabel) {
+                                await this.azureController.getTokens(
+                                    profile, this.accountStore, providerSettings.resources.databaseResource);
+
+                            } else {
+                                return undefined;
```

Comment:
I think we should be rejecting the promise or throw an error instead of returning undefined connectionInfo